src/main/java/org/bigredbands/mb/views/FieldView.java

```diff
@@ -0,0 +1,325 @@
+package org.bigredbands.mb.views;
+
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Shape;
+import java.awt.font.GlyphVector;
+import java.awt.geom.AffineTransform;
+import java.awt.geom.Arc2D;
+import java.awt.geom.Line2D;
+import java.awt.geom.Path2D;
+import java.awt.geom.QuadCurve2D;
+import java.awt.geom.Rectangle2D;
+import java.util.HashMap;
+
+import javax.swing.JPanel;
+
+import org.bigredbands.mb.models.Field;
+import org.bigredbands.mb.models.Point;
+import org.bigredbands.mb.models.RankPosition;
+
+/**
+ * This class represents the generic "field view" used in rendering the editor,
+ * move thumbnails, and exported PDF.
+ */
+public abstract class FieldView extends JPanel {
+
+    protected final Field field;
+    private FieldStyle fieldStyle;
+
+    protected FieldView(Field field, FieldStyle style) {
+        // Set the field information
+        this.field = field;
+        this.fieldStyle = style;
+    }
+
+    protected FieldView(Field field) {
+        this(field, new FieldStyle());
+    }
+
+    public abstract HashMap<String, RankPosition> getRankPositions();
+
+    public void setFieldStyle(FieldStyle fieldStyle) {
+        this.fieldStyle = fieldStyle;
+    }
+
+    // Override getPreferredSize to preserve fixed aspect ratio for the image
+    @Override
+    public Dimension getPreferredSize() {
+        float scaleFactor = getScaleFactor(super.getPreferredSize());
+        return new Dimension((int)(scaleFactor * field.TotalLength),
+                             (int)(scaleFactor * field.TotalHeight));
+    }
+
+    @Override
+    public void paintComponent(Graphics g) {
+        super.paintComponent(g);
```

Comment:
Should this happen after we set the rendering details below?

Reply:
I don't want to side-effect the `super` call (in this case, I think just "standard `JPanel`") with the rendering settings here, and instead only to apply them to the drill panel and any subcomponents (the field, ranks, etc.).

Reply:
Oh, I misread what this method was doing.  Feel free to ignore this comment.